csharp/src/Apache.Arrow/Arrays/Array.cs

```diff
@@ -60,6 +60,20 @@ internal static void Accept<T>(T array, IArrowArrayVisitor visitor)
             }
         }
 
+        public Array Slice(int offset, int length)
+        {
+            if (offset > Length)
```

Comment:
What about when `offset == Length`? Is that valid?

Reply:
Yup in C++ lib